editor/components/post-permalink/index.js

```diff
@@ -8,61 +8,88 @@ import { connect } from 'react-redux';
  */
 import { Component } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
-import { Dashicon, ClipboardButton, Button } from '@wordpress/components';
+import { Dashicon, Button, ClipboardButton, Tooltip } from '@wordpress/components';
 
 /**
  * Internal Dependencies
  */
 import './style.scss';
-import { isEditedPostNew, getEditedPostAttribute } from '../../store/selectors';
+import PostPermalinkEditor from './editor.js';
+import { isEditedPostNew, isPermalinkEditable, getEditedPostPreviewLink, getSamplePermalink } from '../../store/selectors';
+import { getWPAdminURL } from '../../utils/url';
 
 class PostPermalink extends Component {
 	constructor() {
 		super( ...arguments );
+
 		this.state = {
-			showCopyConfirmation: false,
+			iconClass: '',
+			editingPermalink: false,
 		};
-		this.onCopy = this.onCopy.bind( this );
-		this.onFinishCopy = this.onFinishCopy.bind( this );
-	}
-
-	componentWillUnmount() {
-		clearTimeout( this.dismissCopyConfirmation );
 	}
 
-	onCopy() {
-		this.setState( {
-			showCopyConfirmation: true,
-		} );
-	}
-
-	onFinishCopy() {
-		this.setState( {
-			showCopyConfirmation: false,
-		} );
+	componentDidUpdate( prevProps, prevState ) {
+		// If we've just stopped editing the permalink, focus on the new permalink.
+		if ( prevState.editingPermalink && ! this.state.editingPermalink ) {
+			const permalinkButton = document.querySelector( '.editor-post-permalink__link' );
+			if ( permalinkButton ) {
+				permalinkButton.focus();
```

Comment:
The fact that we focus the URL button highlights an (existing) issue with the fade being unnecessary on its focus effect border:

![image](https://user-images.githubusercontent.com/1779930/38060436-b9cd3802-32b8-11e8-818b-482fc30a5af7.png)

Reply:
It'd be nice if we could switch to `text-overflow: fade` instead of the `long-content-fade` mixin.

But in the mean time, adding an `$edge: 1px` parameter, so it's inside the border, fixes this.